File: mobile/android/app/src/main/java/com/example/seniordesignmobileapp/model/AggregatorUiState.kt
package com.example.seniordesignmobileapp.model

enum class BleConnectionPhase(val label: String) {
    IDLE("Idle"),
    WAITING_FOR_PERMISSIONS("Waiting for permissions"),
    BLUETOOTH_UNAVAILABLE("Bluetooth unavailable"),
    BLUETOOTH_DISABLED("Bluetooth disabled"),
    SCANNING("Scanning"),
    CONNECTING("Connecting"),
    REQUESTING_MTU("Requesting MTU"),
    DISCOVERING_SERVICES("Discovering services"),
    SUBSCRIBING("Subscribing"),
    READING_STATUS("Reading status"),
    STREAMING("Streaming"),
    RECOVERING("Recovering"),
    DISCONNECTED("Disconnected"),
}

data class AggregatorUiState(
    val connectionState: String = "Waiting for Bluetooth permissions",
    val connectionPhase: BleConnectionPhase = BleConnectionPhase.WAITING_FOR_PERMISSIONS,
    val deviceName: String = "BLE-Aggregator",
    val isScanning: Boolean = false,
    val isConnected: Boolean = false,
    val latestSample: ImuSample? = null,
    val networkStatus: NetworkStatus? = null,
    val lastSampleHex: String? = null,
    val lastStatusHex: String? = null,
    val errorMessage: String? = null,
    val lastFailureReason: String? = null,
    val reconnectCount: Int = 0,
    val recentEvents: List<String> = emptyList(),
)

File: mobile/ios/FH07Mobile/Model/AggregatorUiState.swift
import Foundation

enum BleConnectionPhase: String, Equatable {
    case idle
    case waitingForPermissions
    case bluetoothUnavailable
    case bluetoothDisabled
    case scanning
    case connecting
    case requestingMtu
    case discoveringServices
    case subscribing
    case readingStatus
    case streaming
    case recovering
    case disconnected

    var label: String {
        switch self {
        case .idle: return "Idle"
        case .waitingForPermissions: return "Waiting for permissions"
        case .bluetoothUnavailable: return "Bluetooth unavailable"
        case .bluetoothDisabled: return "Bluetooth disabled"
        case .scanning: return "Scanning"
        case .connecting: return "Connecting"
        case .requestingMtu: return "Requesting MTU"
        case .discoveringServices: return "Discovering services"
        case .subscribing: return "Subscribing"
        case .readingStatus: return "Reading status"
        case .streaming: return "Streaming"
        case .recovering: return "Recovering"
        case .disconnected: return "Disconnected"
        }
    }
}

struct SavedSessionSummary: Equatable, Identifiable {
    let fileName: String
    let absolutePath: String
    let sizeBytes: Int64
    let modifiedAtEpochMs: Int64

    var id: String { absolutePath }
}

struct AggregatorUiState: Equatable {
    var connectionState: String = "Waiting for Bluetooth permissions"
    var connectionPhase: BleConnectionPhase = .waitingForPermissions
    var deviceName: String = "BLE-Aggregator"
    var isScanning: Bool = false
    var isConnected: Bool = false
    var latestSample: ImuSample? = nil
    var networkStatus: NetworkStatus? = nil
    var lastSampleHex: String? = nil
    var lastStatusHex: String? = nil
    var lastSampleReceivedAtElapsedMs: Int64? = nil
    var lastStatusReceivedAtElapsedMs: Int64? = nil
    var isRecording: Bool = false
    var recordingSessionName: String? = nil
    var recordingSessionPath: String? = nil
    var recordingStartedAtElapsedMs: Int64? = nil
    var recordedSampleCount: Int = 0
    var recordedStatusCount: Int = 0
    var savedSessions: [SavedSessionSummary] = []
    var recordingErrorMessage: String? = nil
    var analysis: AnalysisUiState = AnalysisUiState()
    var modeling: ModelingUiState = ModelingUiState()
    var errorMessage: String? = nil
    var lastFailureReason: String? = nil
    var reconnectCount: Int = 0
    var recentEvents: [String] = []
}
